### DISCUSSION TITLE
Adding defaults for Number helper

### DISCUSSION THREAD

**Discussion by raditzfarhan** [laravel/framework#49254] [Ideas]
## My Problem
Right now, if I want to use `currency` apart from `USD`, I need to specify the currency like so `Number::currency(1000, in: 'MYR')` every time I use it. This can quickly become a hassle if my application will only use that one currency.  Below is the current code snippet with `USD` as the default currency.

```php
/**
     * Convert the given number to its currency equivalent.
     *
     * @param  int|float  $number
     * @param  string  $in
     * @param  ?string  $locale
     * @return string|false
     */
    public static function currency(int|float $number, string $in = 'USD', ?string $locale = null)
    {
        static::ensureIntlExtensionIsInstalled();

        $formatter = new NumberFormatter($locale ?? static::$locale, NumberFormatter::CURRENCY);

        return $formatter->formatCurrency($number, $in);
    }
```

## Proposed Idea
I think it would be nice if we can set a global defaults for `Number` helper like what we already have for `Password` rule. We can set the `defaults` in the `boot` method of the application service provider. So in this way, we can set a default value, not only just for currency, it can also be use for something like locale, or precision. Here is the snippet of how it would look like:-

```php
use Illuminate\Support\Number;

/**
 * Bootstrap any application services.
 */
public function boot(): void
{
    Number::defaults(function () {
        $number = Number::currency('MYR');
 
        return $this->app->isProduction()
                    ? $number->locale('my')
                    : $number;
    });
}
```

## Usage
Then, whenever I call the `currency` method, it will use back my default setting that I have set in application service provider `boot` method without the need for me to specify the currency each time. My code will become a lot shorter and if i want to deploy my app to a different country, I can just change the currency value in the `defaults` without the need to touch the code anywhere else.

```php
Number::currency(1000); // results in  MYR 1,000.00
```

What do you guys think? Any feedbacks or suggestions are welcomed.

Thanks!

**Comment by shaedrich** [comment]
Shouldn't this be ideally in `config/app.php` like the locale and such?

**Reply by raditzfarhan** [reply]
Yeah it could be. But currently the Number helper doesn't seem to be affected by the config parameter like locale. Besides, the currency or precision are not present in the `config/app.php` and I am not sure whether it should be included in the  `config/app.php` or not.

**Reply by shaedrich** [reply]
what would be the harm in doing so? There's still the option to overwrite it in a function call. But doing so seems to be [the least surprise](https://en.wikipedia.org/wiki/Principle_of_least_astonishment) imho.

**Reply by raditzfarhan** [reply]
Yeah I can see your point. Would be great to have this feature.

**Comment by qwertz182** [comment]
What about setting a default currency just like setting a default locale? Like this:

```php
use Illuminate\Support\Number;
use Illuminate\Support\ServiceProvider;
 
class AppServiceProvider extends ServiceProvider
{
    public function boot()
    {
        Number::useCurrency('MYR');
    }
}
```
I think this would be a great addition

**Reply by shaedrich** [reply]
Why not use the [config files](https://laravel.com/docs/configuration) for this and maybe set this via an [environment variable](https://www.dotenv.org/)?

**Reply by qwertz182** [reply]
Yes I would do that anyways, but doesn't that mean that I still have to pass it to the currency method every time I use it?

```php
Number::currency(123.45, config('currency.symbol'));
```

Or am I missing something?

**Reply by shaedrich** [reply]
No, why? Shouldn't the method do that internally automatically? What would be the benefits of defaults otherwise?

**Reply by qwertz182** [reply]
Yeah, but that actually the problem. The methdod doesn't do it internally automatically, at least from what I could see in the code

**Reply by shaedrich** [reply]
Well, then this should be changed 🤷🏻 I would recommend creating a PR but chances aren't low that it would be rejected.

**Reply by Alex-S-Davies** [reply]
This seems like a great idea. It would probably need withCurrency as well as useCurrency just like what has locale offers.

**Reply by qwertz182** [reply]
I created a [pull request](https://github.com/laravel/framework/pull/50010) for this.
